be/src/runtime/workload_group/workload_group.h

```diff
@@ -82,6 +82,31 @@ class WorkloadGroup : public std::enable_shared_from_this<WorkloadGroup> {
 
     int64_t memory_used();
 
+    int spill_threshold_low_water_mark() const {
+        return _spill_low_watermark.load(std::memory_order_relaxed);
+    }
+    int spill_threashold_high_water_mark() const {
+        return _spill_high_watermark.load(std::memory_order_relaxed);
+    }
+
+    void set_weighted_memory_used(int64_t wg_total_mem_used, double ratio);
+
+    std::unordered_set<std::shared_ptr<MemTrackerLimiter>> get_all_mem_trackers();
+
+    int64_t get_weighted_memory_used() {
+        return _weighted_mem_used.load(std::memory_order_relaxed);
+    }
+
+    void check_mem_used(bool& is_low_wartermark, bool& is_high_wartermark) {
```

Comment:
add const to the function signature.
And do not use reference as return value. Use pointer instead.